### PR TITLE
Tx Modal Overlay

### PR DESCRIPTION
**Pull request by eupston**
## Description

- added modal overlay
- put all tx states/responses in overlay

### CODE REVIEW

File: components/Faucet.tsx [hunk 1]
```diff
@@ -7,29 +7,30 @@ import FaucetButton from "@/components/FaucetButton";
...
+	const [response, setResponse] = useState<TxStatus>();
 
 	const fetchSupplyResponse = async () => {
 		if (!address || !network) return;
-
-		setFetchingResponse(true);
+		setResponse({
+			message: "Retrieving Tokens from the Faucet",
+			status: "in-progress",
+		});
```
Comment (aidan-starke) [inline components/Faucet.tsx:25]: could you specify which token is being sent here?

**Review by aidan-starke — APPROVED** [review]
looks awesome, great work

File: components/Faucet.tsx (hunk 1 above)
Comment (eupston) [reply components/Faucet.tsx:25]: oo yeah good call

**Comment by aidan-starke** [comment]
lgtm!

**Pull request merged**